src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -233,6 +233,13 @@ protected function finalizeValue($value)
             throw new UnsetKeyException($msg);
         }
 
+        if ($this->isPlaceholder()) {
```

Comment:
instead of this "isPlaceholder", I would suggest adding `protected function allowPlaceholders()`
would return true in base node
overriden to return false in ArrayNode & co.
would that work?